antarest/study/web/study_data_blueprint.py

```diff
@@ -1020,6 +1021,51 @@ def remove_constraint_term(
             study, binding_constraint_id, term_id
         )
 
+    @bp.get(
+        path="/studies/{uuid}/areas/{area_id}/hydro/allocation",
+        tags=[APITag.study_data],
+        summary="Get allocation options for a given area",
+        response_model_exclude_none=True,
+    )
+    def get_allocation_form_values(
+        uuid: str,
+        area_id: str,
+        current_user: JWTUser = Depends(auth.get_current_user),
+    ) -> AllocationFormFields:
+        params = RequestParameters(user=current_user)
+        study = study_service.check_study_access(
+            uuid, StudyPermissionType.READ, params
+        )
+        all_areas = study_service.get_all_areas(
+            uuid, area_type=AreaType.AREA, ui=False, params=params
+        )
+        return study_service.allocation_manager.get_field_values(
+            study, area_id, all_areas
+        )
+
+    @bp.put(
+        path="/studies/{uuid}/areas/{area_id}/hydro/allocation",
+        tags=[APITag.study_data],
+        summary="Set allocation options for a given area",
+        response_model_exclude_none=True,
+    )
+    def set_allocation_form_values(
+        uuid: str,
+        area_id: str,
+        fields: AllocationFormFields,
```

Comment:
`field_values` for consistance

Reply:
each element in the list represents an `AllocationField` object with an `area_id` and a `coefficient`. `field_values` might be misleading since it suggests that the parameter contains only values. A more appropriate parameter name might be `allocation_fields` or `allocation_data` I used `fields` to keep it more concise, `data` could be an alternative

Reply:
AllocationFormFields represent all the fields 
=> it contains "allocation" field only
=> its value is an object
=> so field_values is logic